corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -0,0 +1,117 @@
+            });
+
+            const draw = new MapboxDraw({
+                // API: https://github.com/mapbox/mapbox-gl-draw/blob/main/docs/API.md
+                displayControlsDefault: false,
+                boxSelect: true, // enables box selection
+                controls: {
+                    polygon: true,
+                    trash: true
+                },
+            });
+            map.addControl(draw);
+
+            map.on('draw.create', updateArea);
+            map.on('draw.delete', updateArea);
+            map.on('draw.update', updateArea);
+
+            function getCoordinates(event) {
+                return event.lngLat;
+            };
+
+            function updateArea(e) {
+                const data = draw.getAll();
+                const area = turf.area(data);
+                // Restrict the area to 2 decimal points.
+                const rounded_area = Math.round(area * 100) / 100;
+            }
+
+            // We should consider refactoring and splitting the below out to a new JS file
+            let clickedMarker;
+
+            self.addCaseMarkersToMap = function (cases) {
+                const markerColor = "#00FF00";
+                cases.forEach(element => {
+                    let coordinates = element.coordinates;
+                    if (coordinates && coordinates.lat && coordinates.lng) {
+                        self.addMarker(coordinates, markerColor);
+                    }
+                });
+            };
+
+            self.addMarker = function (coordinates, color) {
+                const marker = new mapboxgl.Marker({color: color, draggable: false});
+                marker.setLngLat(coordinates);
+                marker.addTo(map);
+            }
+
+            function moveMarkerToClickedCoordinate(coordinates) {
+                if (clickedMarker != null) {
+                    clickedMarker.remove();
+                }
+                if (draw.getMode() === 'draw_polygon') {
+                    // It's weird moving the marker around with the ploygon
+                    return;
+                }
+                clickedMarker = new mapboxgl.Marker({color: "FF0000", draggable: true});
+                clickedMarker.setLngLat(coordinates);
+                clickedMarker.addTo(map);
+            }
+
+            function showCreateCasePopup(coordinates) {
+                new mapboxgl.Popup()
+                .setLngLat(coordinates)
+                .setHTML(coordinates + '<br/><button> Create Case at this point </button>')
+                .addTo(map);
+            }
+
+            // Handle click events here
+            map.on('click', (event) => {
+                let coordinates = getCoordinates(event);
+                moveMarkerToClickedCoordinate(coordinates);
+            });
+            return self;
+        };
+
+        var map;
+
+        $(document).ajaxComplete(function () {
```

Comment:
A comment here will be useful of what this means (it means the report data is loaded?)

Reply:
[255282d](https://github.com/dimagi/commcare-hq/pull/33020/commits/255282dae4f3c1602f1bc76cc39a211dae8bf54e)